src/app/clusters/resource-monitoring-server/replacement-product-list-manager.h

```diff
@@ -39,10 +39,7 @@ class ReplacementProductListManager
 
     void Reset() { mIndex = 0; }
 
-    // Returns total size of Replacement Products List.
-    virtual uint8_t Size() = 0;
-
-    virtual CHIP_ERROR Next(Attributes::ReplacementProductStruct::Type & item) = 0;
+    virtual CHIP_ERROR Next(Attributes::GenericReplacementProductStruct::GenericType & item) = 0;
```

Comment:
This needs to:

1. Document how the callee indicates "end of list" properly.
2. Document what the memory ownership contract is around the outparam.